### PR TITLE
DOC: Clearly state how 2x2 input arrays are handled in stats.linregress

### PR DESCRIPTION
**Pull request by ludcila**
Reference https://github.com/scipy/scipy/issues/9381

### CODE REVIEW

**Comment by ilayn** [comment]
I would prefer the 2x2 mentioned in the notes section but probably a stats resident would judge better. 

The failure is unrelated. If everybody is happy with it, it can go in

**Comment by ilayn** [comment]
@ev-br @WarrenWeckesser do you mind having a quick look at this?

File: scipy/stats/_stats_mstats_common.py [hunk 1]
```diff
@@ -22,6 +22,8 @@ def linregress(x, y=None):
         If only x is given (and y=None), then it must be a two-dimensional
         array where one dimension has length 2.  The two sets of measurements
         are then found by splitting the array along the length-2 dimension.
+        In the case where y=None and x is a 2x2 array, the splitting will be
+        done along the rows, resulting in the sets x[0] and x[1].
```
Comment (ev-br) [inline scipy/stats/_stats_mstats_common.py]: Nit: Maybe rephrase: ... along the rows, i.e., `linregress(x)` is equivalent to `linregress(x[0], x[1])`.

Also, I'd suggest an immediately obvious example, e.g.
```
>>> x = np.array([[0, 1], [0, 2]])
>>> r = linregress(x)
>>> r.slope, r.intercept
(2, 0)
```

Comment (ludcila) [reply scipy/stats/_stats_mstats_common.py]: Thanks @ev-br. Do prefer that line in the parameters section or in the notes?
Also, would that example be the last one in the Examples section?

Comment (ev-br) [reply scipy/stats/_stats_mstats_common.py]: `Notes` are too deep down, I'd put the phrase right there when tge parameter is first introduced.

**Comment by rgommers** [comment]
This was in good shape, I just rebased to trigger CI again and made a few tiny formatting fixes. All green now, so merging. Thanks @ludcila!
